### PR TITLE
fix: Community templates UX issue

### LINKED ISSUE #29165: Community template publish flow UI issues
[Product feedback](https://www.notion.so/appsmith/Project-Review-bc858819eba748a78d14b929e3550505?pvs=4)

Based on this feed back fix issues with:
1. Use cases select - scroll, search, tag visibility
2. Disable the application settings - this will remain as FYI till we updated the design in next iteration 
3. Add a backend event for community templates

### PR DESCRIPTION
**Pull request by rahulbarwal**
## Description
This pull request includes several fixes and improvements to the codebase. Here is a summary of the changes:

- Improved searchability and tag visibility for the application.

- Disabled the public and forkable switches in ApplicationSettings.

- Refactored the CommunityTemplateForm to use the defaultPageId.

- Added a new event, COMMUNITY_TEMPLATE_PUBLISHED.

#### PR fixes following issue(s)
Fixes #29165

#### Type of change
- Bug fix (non-breaking change which fixes an issue)
#### How Has This Been Tested?
- [x] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **New Features**
  - Introduced the ability to publish community templates with enhanced settings for visibility and collaboration.
  - Added new analytics event tracking for community template publication.

- **Enhancements**
  - Improved the search functionality in the template selection with responsive tag count and search feature.
  - Enhanced application settings with the ability to disable switches for public and forkable options.

- **Documentation**
  - Updated instructions for triaging new changes and features.

- **Refactor**
  - Simplified logic for determining default page ID in the template publishing form.

- **Tests**
  - Extended unit tests to cover new session user service functionality.

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ApplicationTemplateServiceCEImpl.java [hunk 1]
```diff
@@ -54,6 +56,7 @@ public class ApplicationTemplateServiceCEImpl implements ApplicationTemplateServ
     private final ApplicationService applicationService;
     private final ResponseUtils responseUtils;
     private final ApplicationPermission applicationPermission;
+    private final SessionUserService sessionUserService;
 
     private final ObjectMapper objectMapper;
```
[the remark this answers is not in the document]
Comment (AnaghHegde) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ApplicationTemplateServiceCEImpl.java:61]: @rahulbarwal This does make sense. Can you make this change?

[the remark this answers is not in the document]
Comment (rahulbarwal) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ApplicationTemplateServiceCEImpl.java:61]: @AnaghHegde this variable is declared before the constructor. Not sure what it means?
Should I remove the extra line that is added between this and other declrations?

[the remark this answers is not in the document]
Comment (AnaghHegde) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ApplicationTemplateServiceCEImpl.java:61]: Okay so what it means is that the order is not correct. The newly added SessionUserService should be after the object mapper.

**Review by jacquesikot — APPROVED** [review]

**Review by AnaghHegde — APPROVED** [review]

**Pull request merged**